rfcs/rfc-create-ui-architecture.md

```diff
@@ -0,0 +1,183 @@
+# RFC for a "Create UI" Architecture To Support New Filing Types
+
+- Start Date: (2021-03-10)
+- Target Major Version: None
+- Zenhub Ticket #: https://github.com/bcgov/entity/issues/6725
+
+## Introduction
+
+Currently, the [Create UI web app](https://github.com/bcgov/bcrs-business-create-ui
+) allows a user to enter and file a Benefit Company (BEN) Incorporation Application (IA). This web app needs to be enhanced to support many other types of incorporation and continuation filings.
+
+The purpose of this RFC is to propose/communicate/discuss/plan the implementation of an architecture in this web app to support these other filings. The goals of this architecture are to:
+
+- speed up the integration of new types of filings
+- promote easy-to-understand code flows as they relate to different filing types
+- promote maximum reuse of existing architecture, components and other code
+- facilitate maintenance of increasingly-complex code
+- minimize large future refactors
+- apply best practices learned in the development of other BCROS web apps (and other apps in general)
+
+## Summary
+
+The requirement is for a way to capture business logic in the code, but have it easy to work with.
+
+The proposed solution is two-fold:
+
+1. centralize the differences of the various filing types in one place -- using a "resources" (aka config) object that contains data specific to each filing type
+
+2. update the code to function differently, where needed, according to each filing's configuration
+
+Note: This discussion excludes the integration of a high-level rules engine such as [json-rules-engine](https://www.npmjs.com/package/json-rules-engine). However, we may be able to copy its syntax and terminology -- to be looked at during detailed design of our config object.
+
+## Detailed Solution
+
+### 1. Resources (Config) Object
```

Comment:
do we have a feel for the grouping of these _concerns_?
for me I see them in sort of 3 buckets:
- the order of and which routes (business info, directors, shares, pay)
- the rules that the components need (date values between, region == 'BC')
- and the text used in the components

Reply:
I agree with your buckets. Do you think the concerns need to be groups? I am hoping the config object could contain all of those (or references to child files).

Reply:
I think a single wrapper holds it all, it just might make it easier to grok and change if say all the routing was together, than spread across the fields ... although that could make sense too, it each item had everything in a step together.

the RFC should have a sample design, eg
https://github.com/vuejs/composition-api-rfc/blob/master/index.md